indico/modules/events/timetable/client/js/DayTimetable.tsx

```diff
@@ -38,23 +38,33 @@ interface DayTimetableProps {
   entries: TopLevelEntry[];
 }
 
-function TopLevelEntries({dt, entries}: {dt: Moment; entries: TopLevelEntry[]}) {
+function TopLevelEntries({
+  dt,
+  entries,
+  eventId,
+}: {
+  dt: Moment;
+  entries: TopLevelEntry[];
+  eventId: number;
+}) {
   const dispatch = useDispatch();
 
   const setDurations = useMemo(() => {
     const obj = {};
     for (const e of entries) {
       obj[e.id] = (duration: number) =>
-        dispatch(actions.resizeEntry(dt.format('YYYYMMDD'), e.id, duration));
+        dispatch(actions.resizeEntry(e, eventId, duration, dt.format('YYYYMMDD')));
     }
     return obj;
-  }, [entries, dispatch, dt]);
+  }, [entries, dispatch, dt, eventId]);
 
   const setChildDurations = useMemo(() => {
     const obj = {};
     for (const e of entries) {
-      obj[e.id] = (id: string) => (duration: number) =>
-        dispatch(actions.resizeEntry(dt.format('YYYYMMDD'), id, duration, e.id));
+      obj[e.id] = (childId: string) => (duration: number) => {
```

Comment:
I'm being annoying here but we could use a reducer instead of the separate const obj and for loop etc